protocol/bc/map.go

```diff
@@ -76,10 +74,36 @@ func mapTx(tx *bc.TxData) (headerID bc.Hash, hdr *header, entryMap map[bc.Hash]e
 				}
 
 				assetID := oldIss.AssetID()
-				b := vmutil.NewBuilder()
-				b = b.AddData(oldIss.Nonce).AddOp(vm.OP_DROP).AddOp(vm.OP_ASSET).AddData(assetID[:]).AddOp(vm.OP_EQUAL)
 
-				nonce = newNonce(program{1, b.Program}, tr)
+				// This is the program
+				//   [PUSHDATA(oldIss.Nonce) DROP ASSET PUSHDATA(assetID) EQUAL]
+				// minus a circular dependency on protocol/vm.
```

Comment:
This reminds me I would like to get rid of the vm -> bc dependency. Package vm should instead define one or two small interfaces that bc satisfies. But this seems fine to me for now.